plugins/odpcache/services/in_memory_cache.go

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ * Copyright 2023, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package services //
+package services
+
+import (
+	"time"
+
+	"github.com/optimizely/agent/plugins/odpcache"
+	"github.com/optimizely/go-sdk/pkg/odp/cache"
+)
+
+// InMemoryCache represents the in-memory implementation of Cache interface
+type InMemoryCache struct {
```

Comment:
We are now using this as default odp cache with both `timeout` and `size` as arguments. Following is the default configuration:

```
odp:
      ## Disable odp
      disable: false
      ## Timeout in seconds after which event requests will timeout.
      eventsRequestTimeout: 10s
      ## Flush interval in seconds for odp events
      eventsFlushInterval: 1s
      ## Timeout in seconds after which segment requests will timeout.
      segmentsRequestTimeout: 10s
      ## This will override all other settings
      segmentsCache:
        default: "in-memory"
        services:
          in-memory: 
            size: 10000
            timeout: 600 # seconds
        #   redis: 
        #     host: "localhost:6379"
        #     password: ""
        #     database: 0
        #     timeout: 0 # seconds
```